tools/serverpod_cli/lib/src/generator/dart/server_code_generator.dart

```diff
@@ -33,14 +34,26 @@ class DartServerCodeGenerator extends CodeGenerator {
         ModelFilterUtil.getClassesWithoutSealedHierarchies(models);
 
     for (var sealedHierarchy in sealedHierarchies) {
-      var partOfAllocator = PartOfAllocator([]);
-      var partAllocator = PartAllocator(partOfAllocator);
+      var topNode =
```

Comment:
Question/Discussion: This code needs to be kept in sync between the client and the server. Maybe it is time to break it out into a helper that manages this? Seems unecessary to repeate the code